applications/CSharpWrapperApplication/kratos_internals.cpp

```diff
@@ -1,72 +1,235 @@
-	pmMainModelPart->GetProperties(0).SetValue(Kratos::CONSTITUTIVE_LAW, pCl);
-	pmMainModelPart->SetBufferSize(2);
+    // Creating model part
+    auto& r_model_part = mModel.CreateModelPart(mModelpartName, buffer_size);
+    r_model_part.GetProcessInfo()[Kratos::DOMAIN_SIZE] = domain_size;
 
-	Kratos::VariableUtils().AddDofWithReaction(Kratos::DISPLACEMENT_X, Kratos::REACTION_X, *pmMainModelPart);
-	Kratos::VariableUtils().AddDofWithReaction(Kratos::DISPLACEMENT_Y, Kratos::REACTION_Y, *pmMainModelPart);
-	Kratos::VariableUtils().AddDofWithReaction(Kratos::DISPLACEMENT_Z, Kratos::REACTION_Z, *pmMainModelPart);
+    // Basic variables
+    r_model_part.AddNodalSolutionStepVariable(Kratos::DISPLACEMENT);
+    r_model_part.AddNodalSolutionStepVariable(Kratos::REACTION);
+    r_model_part.AddNodalSolutionStepVariable(Kratos::VOLUME_ACCELERATION);
 
-	pmMainModelPart->CreateSubModelPart(SKIN_SUBMODEL_PART_NAME);
+    // Adding custom variables
+    const std::size_t n_variables = mSettingsParameters["solver_settings"]["auxiliary_variables_list"].size();
+    for (std::size_t i_var = 0; i_var < n_variables; ++i_var) {
+        const std::string& r_variable_name = mSettingsParameters["solver_settings"]["auxiliary_variables_list"].GetArrayItem(i_var).GetString();
+        if (Kratos::KratosComponents<Kratos::Variable<double>>::Has(r_variable_name)){
+            const auto& r_variable = Kratos::KratosComponents<Kratos::Variable<double>>::Get(r_variable_name);
+            r_model_part.AddNodalSolutionStepVariable(r_variable);
+        } else if (Kratos::KratosComponents<Kratos::Variable<Kratos::array_1d<double, 3>>>::Has(r_variable_name)) {
+            const auto& r_variable = Kratos::KratosComponents<Kratos::Variable<Kratos::array_1d<double, 3>>>::Get(r_variable_name);
+            r_model_part.AddNodalSolutionStepVariable(r_variable);
+        }
+    }
+
+    // NOTE: This is hardcoded
+    r_model_part.CreateSubModelPart(SKIN_SUBMODEL_PART_NAME);
+}
+
+void KratosInternals::loadMDPA(const std::string& rMDPAFilePath) {
+    // Getting model part
+    auto& r_model_part = GetMainModelPart();
+
+    // Reading file
+    Kratos::shared_ptr<std::fstream> pFile = Kratos::make_shared<std::fstream>();
+    pFile->open(rMDPAFilePath, std::fstream::in);
+    Kratos::ModelPartIO(pFile).ReadModelPart(r_model_part);
+    pFile->close();
+}
+
+void KratosInternals::loadSettingsParameters(const std::string& rJSONFilePath) {
+    if (rJSONFilePath != "") {
+        std::ifstream infile(rJSONFilePath);
+        if (infile.good()) std::cout << "JSON file: " << rJSONFilePath  << " cannot be found" << std::endl;
+        std::stringstream buffer;
+        buffer << infile.rdbuf();
+        mSettingsParameters = Kratos::Parameters(buffer.str());
+    }
+
+    Kratos::Parameters default_parameters = GetDefaultSettings();
+    mSettingsParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
```

Comment:
Why validate? A few keys are missing from default parameters, like solver_type
```suggestion
    mSettingsParameters.RecursivelyAddMissingParameters(default_parameters);
```

Reply:
The same procedure follow in the mechanical solver is follow, the solver type is checked later

Reply:
"If the parameters contain values that do not appear in the defaults, an error is thrown" If I understant correctly, validation will throw an error when, for example, solver type is defined in the loaded parameters json. Isn't that right?

Reply:
Huuum, yes, now I remember we do some additional on the python script. One sec

Reply:
Yes indeed, you are right